packages/stellar-axelar-std-derive/src/operatable.rs

```diff
@@ -17,3 +18,34 @@ pub fn operatable(name: &Ident) -> TokenStream2 {
         }
     }
 }
+pub fn only_operator_impl(input_fn: ItemFn) -> TokenStream2 {
+    let fn_vis = &input_fn.vis;
+    let fn_sig = &input_fn.sig;
+    let fn_name = &fn_sig.ident;
+    let fn_generics = &fn_sig.generics;
+    let fn_inputs = &fn_sig.inputs;
+    let fn_output = &fn_sig.output;
+    let fn_body = &input_fn.block;
+    let fn_attrs = &input_fn.attrs;
+
+    // Check that env is the first parameter
+    let Some(syn::FnArg::Typed(pat_type)) = fn_inputs.first() else {
+        panic!("First parameter must be a typed parameter")
+    };
+    let syn::Pat::Ident(pat_ident) = &*pat_type.pat else {
+        panic!("First parameter must be a simple identifier")
+    };
+    assert!(
+        pat_ident.ident == "env",
+        "First parameter must be named 'env'"
+    );
+
+    quote! {
+        #(#fn_attrs)*
+        #fn_vis fn #fn_name #fn_generics(#fn_inputs) #fn_output {
+            Self::operator(&env).require_auth();
```

Comment:
All 3 functions are mostly a duplicate. Create a `modifier_impl` helper that does all of this except taking one arg, which is this statement that needs to differ between the 3 implementations. Makes it easier to add more modifiers like these in the future

Reply:
Failure tests will also just be needed once (instead of for every modifier/macro)